python/paddle/distributed/auto_parallel/static/pir_pass.py

```diff
@@ -104,24 +121,28 @@ def apply_partition_pass(program):
             for op in lr_ops[1:]:
                 lr = op.result(0)
                 lr.replace_all_uses_with(lr_value)
-                new_program.global_block().remove_op(op)
+                program.global_block().remove_op(op)
 
-    return new_program
+        pm = paddle.pir.PassManager()
+        pm.add_pass(
+            'dead_code_elimination_pass', {}
+        )  # apply pass to eliminate dead code
+        pm.run(program)
```

Comment:
as discussed, do not apply 'dead_code_elimination_pass' before add fetch ops.

Reply:
done, thanks!